apollo-runtime/src/main/java/com/apollographql/apollo/internal/RealApolloSubscriptionCall.java

```diff
@@ -0,0 +1,178 @@
+package com.apollographql.apollo.internal;
+
+import com.apollographql.apollo.ApolloSubscriptionCall;
+import com.apollographql.apollo.api.Response;
+import com.apollographql.apollo.api.Subscription;
+import com.apollographql.apollo.exception.ApolloCanceledException;
+import com.apollographql.apollo.exception.ApolloNetworkException;
+import com.apollographql.apollo.internal.subscription.ApolloSubscriptionException;
+import com.apollographql.apollo.internal.subscription.SubscriptionManager;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.annotation.Nonnull;
+
+import static com.apollographql.apollo.api.internal.Utils.checkNotNull;
+import static com.apollographql.apollo.internal.CallState.ACTIVE;
+import static com.apollographql.apollo.internal.CallState.CANCELED;
+import static com.apollographql.apollo.internal.CallState.IDLE;
+import static com.apollographql.apollo.internal.CallState.TERMINATED;
+
+public class RealApolloSubscriptionCall<T> implements ApolloSubscriptionCall<T> {
+  private final Subscription<?, T, ?> subscription;
+  private final SubscriptionManager subscriptionManager;
+  private final AtomicReference<CallState> state = new AtomicReference<>(IDLE);
+  private SubscriptionManagerCallback<T> subscriptionCallback;
+
+  public RealApolloSubscriptionCall(Subscription<?, T, ?> subscription, SubscriptionManager subscriptionManager) {
+    this.subscription = subscription;
+    this.subscriptionManager = subscriptionManager;
+  }
+
+  @Override
+  public void start(@Nonnull Callback<T> callback) throws ApolloCanceledException {
```

Comment:
is `start` standard? thoughts on `connect` or `subscribe`?

Reply:
`subscribe` / `unsubscribe` then?

Reply:
any idea what apollo js calls it?

Reply:
looking into https://github.com/apollographql/apollo-client/blob/master/docs/source/features/subscriptions.md

seems it's called `subscribeToMore`

Reply:
but if we call it subscribe then our API will look like this:
```
 application.apolloClient()
        .subscribe(new RepoCommentAddedSubscription(repoFullName))
        .subscribe(new ApolloSubscriptionCall.Callback<RepoCommentAddedSubscription.Data>() {
          @Override public void onResponse(@Nonnull Response<RepoCommentAddedSubscription.Data> response) {
            Log.d("Subscription!!!", response.data().toString());
          }

          @Override public void onFailure(@Nonnull ApolloException e) {
            Log.e("Subscription!!!", "", e);
          }

          @Override public void onCompleted() {
            Log.d("Subscription!!!", "Completed!");
          }
        });
```

where first subscribe returns `ApolloSubscriptionCall` and second is actually start subscription